api/routes/authRoutes.js

```diff
@@ -0,0 +1,51 @@
+const express = require('express');
+const router = express.Router();
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const { jwtSecret } = require('../../config/secrets');
+const { getUserByUsername, addUser } = require('../../data/helpers');
+
+router.post('/login', async (req, res) => {
+  try {
+    const { username, password } = req.body;
+    if (!username || !password)
+      return res.status(400).json({
+        error: 'Request must include values for username and password.',
+      });
+    const [user] = await getUserByUsername(username);
+    if (!user || !bcrypt.compareSync(password, user.password))
+      throw new Error();
+    const token = jwt.sign({ id: user.id }, jwtSecret, { expiresIn: '1d' });
+    return res.json({ token });
```

Comment:
I suggest in addition to the token, you should a property like `isProvider` or something. When the frontend receives the response, they can read that property, and redirect the user to the correct home page for example.